lbrynet/lbrynet_daemon/LBRYDownloader.py

```diff
@@ -85,27 +86,23 @@ def check_status(self):
 
     def start(self, stream_info, name):
         self.resolved_name = name
-        self.stream_info = stream_info
-        if 'sources' in self.stream_info:
-            self.stream_hash = self.stream_info['sources']['lbry_sd_hash']
-        else:
-            raise InvalidStreamInfoError(self.stream_info)
-        if 'description' in self.stream_info:
-            self.description = self.stream_info['description']
+        self.stream_info = deepcopy(stream_info)
+        self.description = self.stream_info['description']
+        self.stream_hash = self.stream_info['sources']['lbry_sd_hash']
+
         if 'fee' in self.stream_info:
-            if 'LBC' in self.stream_info['fee']:
-                self.key_fee = float(self.stream_info['fee']['LBC']['amount'])
-                self.key_fee_address = self.stream_info['fee']['LBC']['address']
-            else:
-                self.key_fee_address = None
-        else:
-            self.key_fee = None
-            self.key_fee_address = None
-        if self.key_fee > self.max_key_fee:
-            log.info("Key fee %f above limit of %f didn't download lbry://%s" % (self.key_fee, self.max_key_fee, self.resolved_name))
-            return defer.fail(None)
-        else:
-            pass
+            self.fee = LBRYFee(self.stream_info['fee'], {'USDBTC': self.wallet._USDBTC, 'BTCLBC': self.wallet._BTCLBC})
```

Comment:
Following my suggestions from earlier, this would change to:
`self.fee = LBRYFee(self.stream_info['fee'], self.exchange_rates.get_rate_dict())`